lib/partials2melodies.js

```diff
@@ -29,21 +29,21 @@ function partials2melodies(_partials) {
 
     // define subfunctions
     function detectNoteOnNoteOffIndices(amps) {
-      const result = [];
+      const noteOnOffs = [];
       // 'note on' = 1, 'note off' = -1, 'nothing' = 0
       amps.forEach((amp, idx, amps) => {
```

Comment:
I think this whole block can be refactored with `amps.map`, resulting with the elimination of needless `noteOnOffs` variable.

Reply:
Also if you use `amps.map` your nested if-statements will be much cleaner because you can early-return noteOnOff values, which will also get rid of the last else-statement.